### ISSUE TITLE
Getting started: More use cases

### ISSUE THREAD

**Issue by jpike88** [typescript-eslint/typescript-eslint#1298]
I work on OSS projects like bundlers and other toolchain projects. Migrating from tslint to eslint looks insanely complex and after messing around I still haven't got any idea on how to perform migration properly.

Splitting into 5 different packages and requiring people to study an essay is not a good way of onboarding people, the time it takes is way more than it deserves.

A use case that shows the minimal clear steps needed to fire things up is ideal.

In the meantime I'm going to stick with tslint until I don't feel like I'm stabbing in the dark with migration

**Comment by bradzacher** [comment]
Yeah we definitely need a better onboarding documentation.
It's been on the todo list, but ultimately it's hard 

In the meantime - the best documentation is this repo itself.
We are a monorepo, and we use our own tooling to lint our codebase.

Check out our eslint config: https://github.com/typescript-eslint/typescript-eslint/blob/master/.eslintrc.js

installation should be as simple as
```bash
$ yarn add eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin
```

You can ignore or add in all of the other plugins if you desire.

**Comment by PutziSan** [comment]
Would you be open to adding documentation for a quick start on the [first page](https://github.com/typescript-eslint/typescript-eslint/blob/master/README.md)?

```diff
 ## Getting Started
 
+ For a quick start, install the packages `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin`:
+ 
+ ```sh
+ npm install @typescript-eslint/parser @typescript-eslint/eslint-plugin --save-dev
+ # or with yarn
+ yarn add eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin -D
+ ```
+ 
+ And adjust your "eslintrc":
+ 
+ ```json
+ {
+   "extends": ["plugin:@typescript-eslint/recommended"]
+ }
+ ```
+ 
+ `"plugin:@typescript-eslint/recommended"` sets up [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser) and activates the [recommended rules](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#supported-rules).
+ 
 The following sections will give you an overview of what this project is, why it exists and how it works at a high level.
```

The many packages can be very overwhelming for the first use case "I want to check my TypeScript files with eslint". 
At the beginning the documentation of the [eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin) is probably enough for this user.

**Comment by David-Else** [comment]
I agree that setup is far too convoluted and complex. 99% of people just want to NPM install and copy-paste some default settings. Even the copy-paste settings are a potential waste of time and effort, and even now knowing the repo, I still can't easily find the instructions on how to do this hidden inside the labyrinth!

I think the crux of the problem is that the repo is aimed at the coder who wants to contribute and cares about all the parts of the jigsaw, but I imagine that 99% of people visiting are actually looking for https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin and how to set it up.

Thinking of a user journey from https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#usage...

- Why are there 2 separate packages `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin`? Why can't the plugin just include the parser?
- What are all these different config options, it does not make sense, my head is going to explode!

Imagine instead the instructions are:
```
npm install @typescript-eslint/eslint-plugin
type install.sh
```
Welcome to TypeScript eslint plugin! Please select from the following options:

1. Do you use Prettier for your code formatting?
2. Would you like to use the official ESLint recommeded settings?
3. Would you like to use our recommended settings?
4. Would you like to take advantage of our rules that use Types? It is better, but may take longer to process.

Thanks... I have generated a `.eslintrc` config file for you in this directory. Enjoy your linting :)

As for airbnb... I have worked on trying to simplify that for TypeScript users and reduced the entire thing down to one file :) 

https://www.elsewebdevelopment.com/typescript-airbnb-style-guide-optimized-for-prettier-in-one-json-file/

If the entire process could be automated with and without prettier support then it would be a nice way to add airbnb support.

**Comment by bradzacher** [comment]
I've started writing a new onboarding / getting started guide.
Would love some feedback - #1324

Rendered: https://github.com/typescript-eslint/typescript-eslint/blob/onboarding-docs/docs/getting-started/README.md

**Comment by David-Else** [comment]
@bradzacher These are a MASSIVE improvement, really well done and easy for people to understand. Excellent!

**Comment by jpike88** [comment]
I like it too, much better
